stdlib/OpenSSL_jll/src/OpenSSL_jll.jl

```diff
@@ -0,0 +1,65 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+## dummy stub for https://github.com/JuliaBinaryWrappers/OpenSSL_jll.jl
+
+baremodule OpenSSL_jll
+using Base, Libdl, Base.BinaryPlatforms
+using NetworkOptions: ca_roots
+
+const PATH_list = String[]
+const LIBPATH_list = String[]
+
+export libcrypto, libssl
+
+# These get calculated in __init__()
+const PATH = Ref("")
+const LIBPATH = Ref("")
+artifact_dir::String = ""
+libcrypto_handle::Ptr{Cvoid} = C_NULL
+libcrypto_path::String = ""
+libssl_handle::Ptr{Cvoid} = C_NULL
+libssl_path::String = ""
+
+if Sys.iswindows()
+    if arch(HostPlatform()) == "x86_64"
+        const libcrypto = "libcrypto-3-x64.dll"
+        const libssl = "libssl-3-x64.dll"
+    else
+        const libcrypto = "libcrypto-3.dll"
+        const libssl = "libssl-3.dll"
+    end
+elseif Sys.isapple()
+    const libcrypto = "@rpath/libcrypto.3.dylib"
+    const libssl = "@rpath/libssl.3.dylib"
+else
+    const libcrypto = "libcrypto.so.3"
+    const libssl = "libssl.so.3"
+end
+
+function __init__()
+    global libcrypto_handle = dlopen(libcrypto)
+    global libcrypto_path = dlpath(libcrypto_handle)
+    global libssl_handle = dlopen(libssl)
+    global libssl_path = dlpath(libssl_handle)
+    global artifact_dir = dirname(Sys.BINDIR)
+    LIBPATH[] = dirname(libssl_path)
+    push!(LIBPATH_list, LIBPATH[])
+
+    # If not already done, set the environment variable `SSL_CERT_FILE` when necessary.
+    cert_loc = ca_roots()
+    if cert_loc isa String
+        get!(ENV, "SSL_CERT_FILE", cert_loc)
+    end
```

Comment:
@StefanKarpinski can you please check this makes sense?  I followed the same logic as in https://github.com/JuliaLang/julia/blob/50063122b44df1ba80e83fb5abee3f05074f0223/stdlib/LibGit2/src/LibGit2.jl#L1027-L1028 in practice this should be set only on Linux and FreeBSD, unless the user set one of the relevant certificates env variables already.

Reply:
Shouldn't this be done in NetworkOptions though, so that everyone gets it?